### ISSUE TITLE
Constraint violation error returned when creating resource with validated JSON

### ISSUE THREAD

**Issue by cpschult** [IBM/FHIR#1688]
**Describe the bug**
IBM FHIR server v4.4.2 returns an error when creating a resource with JSON known to pass validation.

**To Reproduce**
1. Attempt to create FHIR resource via PUT.  See attached example EOB validated to c4bb 0.1.7.
[example-eob-constraint-violation-c4bb017-json.txt](https://github.com/IBM/FHIR/files/5518063/example-eob-constraint-violation-c4bb017-json.txt)

2. Returned OperationalOutcome shows an error for constraint violation.  Example:
```
"severity": "error", 
                "code": "invariant", 
                "details": {
                    "text": "generated-C4BB-ExplanationOfBenefit-Outpatient-Institutional-19: Constraint violation: item.exists() implies (item.all((revenue.exists() implies (revenue.memberOf('http://hl7.org/fhir/us/carin-bb/ValueSet/AHANUBCRevenueCodes', 'required'))) and productOrService.exists() and productOrService.memberOf('http://hl7.org/fhir/us/carin-bb/ValueSet/AMACPTCMSHCPCSProcedureCodes', 'required') and (modifier.exists() implies (modifier.all(memberOf('http://hl7.org/fhir/us/carin-bb/ValueSet/AMACPTCMSHCPCSModifiers', 'required')))) and (serviced.exists() implies (serviced.as(date).exists())) and adjudication.where(category.exists() and category.memberOf('http://hl7.org/fhir/us/carin-bb/ValueSet/C4BBAdjudication', 'required')).exists() and (adjudication.where(category.where(coding.where(system = 'http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBAdjudicationDiscriminator' and code = 'denialreason').exists()).exists()).exists() implies (adjudication.where(category.where(coding.where(system = 'http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBAdjudicationDiscriminator' and code = 'denialreason').exists()).exists() and reason.exists() and reason.memberOf('http://hl7.org/fhir/us/carin-bb/ValueSet/X12ClaimAdjustmentReasonCodesCMSRemittanceAdviceRemarkCodes', 'required')).exists())) and (adjudication.where(category.where(coding.where(system = 'http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBAdjudicationDiscriminator' and code = 'allowedunits').exists()).exists()).exists() implies (adjudication.where(category.where(coding.where(system = 'http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBAdjudicationDiscriminator' and code = 'allowedunits').exists()).exists()).exists()))))"
                }, 
                "expression": [
                    "ExplanationOfBenefit"
                ]
```

**Expected behavior**
Resource should be created successfully.

**Comment by prb112** [comment]
I believe this is fixed in 4.5.0.

**Comment by JohnTimm** [comment]
Duplicate of issue #1630. Fixed by PR #1632.
